src/vos/vos_tree.c

```diff
@@ -1014,7 +1013,15 @@ key_tree_prepare(struct vos_object *obj, daos_handle_t toh,
  out:
 	D_CDEBUG(rc == 0, DB_TRACE, DB_IO, "prepare tree, flags=%x, tclass=%d %d\n",
 		 flags, tclass, rc);
-	return rc;
+
+	if (rc != 0 || tclass != VOS_BTR_AKEY || !(flags & SUBTR_CREATE))
+		return rc;
+
+	/* As a first cut for aggregation detection, just return 1 if it's not the first update.
+	 * This can be improved upon for evtree to take into account actual extent overlap but
+	 * this is a simpler solution for now.
+	 */
```

Comment:
For non-created case, we will return 1, then for "!(flags & SUBTR_CREATE)", it must not be the first update, then we need to return 1 instead of 0, right?

Reply:
If SUBTR_CREATE isn't set, it's not an update